check if there are any pending DO migrations for a worker

nikitassharma opened the pull request:
Fixes CC-6312

DO migrations block version uploads. Currently, the only reliable way to check if a DO migration is to run `wrangler versions upload` and inspect the error to determine if it's a result of an unsupported DO migration. This is especially important for CI scripting to determine if changes should be deployed instead of uploading a version. Also addresses https://github.com/cloudflare/workers-sdk/issues/10965.

Design decisions made by this pr:
* Uses `wrangler check do-migrations` to avoid confusion with D1 migrations
* This information was not added to a `--dry-run` call to avoid making API calls on a dry run
  * Avoids making the assumption that version uploads will not support DO migrations in the future.


---

<!--
Please don't delete the checkboxes <3
The following selections do not need to be completed if this PR only contains changes to .md files
-->

- Tests
  - [X] Tests included
  - [ ] Tests not necessary because:
- Public documentation
  - [ ] Cloudflare docs PR(s): <!--e.g. <https://github.com/cloudflare/cloudflare-docs/pull/>...-->
  - [X] Documentation not necessary because: `wrangler check` isn't documented
- Wrangler V3 Backport
  - [ ] Wrangler PR: <!--e.g. <https://github.com/cloudflare/workers-sdk/pull/>...-->
  - [X] Not necessary because: not in v3

<!--
Have you read our [Contributing guide](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md)?
In particular, for non-trivial changes, please always engage on the issue or create a discussion or feature request issue first before writing your code.
-->

nikitassharma committed:
add `wrangler check do-migration` command

nikitassharma committed:
add tests for do migration check

nikitassharma committed:
changeset

nikitassharma committed:
prettify

nikitassharma committed:
checks

nikitassharma committed:
fix test

nikitassharma committed:
fixed?

nikitassharma committed:
Merge branch 'main' into nsharma/check-migration

emily-shen commented on packages/wrangler/src/check/migration.ts:
```diff
@@ -0,0 +1,90 @@
+
+export const checkNamespace = createNamespace({
+	metadata: {
+		description: "🔍 Check pending migrations for your Worker",
```
```suggestion
		description: "🔍 Check pending migrations for your Durable Object",
```

emily-shen reviewed (commented):
merge conflicts :(

petebacondarwin commented:
Converted to draft until the rebase conflicts are resolved.

petebacondarwin commented:
Hi @nikitassharma — we're doing some cleanup of stale draft PRs.

This PR has been quiet since 2026-01-07. Could you let us know within **14 days** whether you're still planning to continue?

- If yes: please leave a comment with a status update or push a rebase, and we'll keep it open.
- If no: no need to reply — we'll close it after 14 days, and you can always reopen later if you pick it back up.

Thanks!

petebacondarwin committed:
fixup: import runInTempDir from workers-utils

The local helpers/run-in-tmp.ts wrapper was removed in #13954. New test files
must import runInTempDir directly from @cloudflare/workers-utils/test-helpers.
